packages/upload-api/src/types/rate-limits.ts

```diff
@@ -0,0 +1,43 @@
+import * as Ucanto from '@ucanto/interface'
+
+export type RateLimitID = string
```

Comment:
Could you add a comment explaining what and how it is derived from. As an aside I am not a fan of IDs that you need to keep track of, any reason why remove can't take same `{subject, rate}` pair as key ? It appears a lot more appealing IMO.

Reply:
So my reasoning here was that an implementor might want to allow the same `{subject, rate}` pair in multiple rate limits - more than one party may be in charge of adding rate limits in the future (eg, the fraud prevention department flagging and blocking an account because they detected phishing sites being uploaded and, separately, billing blocking an account for non-payment) and in that case I think they would likely not want the removal of one block to always imply the removal of another block.